akka-http2-support/src/main/scala/akka/http/impl/engine/http2/TlsActorCopy.scala

```diff
@@ -0,0 +1,485 @@
+/**
+ * Copyright (C) 2015-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.impl.io
+
+import java.nio.ByteBuffer
+import javax.net.ssl.SSLEngineResult.HandshakeStatus
+import javax.net.ssl.SSLEngineResult.HandshakeStatus._
+import javax.net.ssl.SSLEngineResult.Status._
+import javax.net.ssl._
+
+import akka.actor._
+import akka.stream._
+import akka.stream.impl.FanIn.InputBunch
+import akka.stream.impl.FanOut.OutputBunch
+import akka.stream.impl._
+import akka.util.ByteString
+
+import scala.annotation.tailrec
+import akka.stream.TLSProtocol._
+
+import scala.util.control.NonFatal
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * INTERNAL API.
+ */
+private[stream] object TLSActor {
```

Comment:
Yep, the copied actor needs two changes two work with the current netty 4.1.6.Final:

```diff
diff --git a/akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala b/akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala
index 3fa96252..01d9373d 100644
--- a/akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala
+++ b/akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala
@@ -131,7 +131,7 @@ private[stream] class TLSActor(
 
   // These are Netty's default values
   // 16665 + 1024 (room for compressed data) + 1024 (for OpenJDK compatibility)
-  private val transportOutBuffer = ByteBuffer.allocate(16665 + 2048)
+  private val transportOutBuffer = ByteBuffer.allocate(16665 + 2048 + 1000)
   /*
    * deviating here: chopping multiple input packets into this buffer can lead to
    * an OVERFLOW signal that also is an UNDERFLOW; avoid unnecessary copying by
@@ -378,7 +378,7 @@ private[stream] class TLSActor(
             handshakeFinished()
             transportInChoppingBlock.putBack(transportInBuffer)
           case _ ⇒
-            if (transportInBuffer.hasRemaining) doUnwrap()
+            if (transportInBuffer.hasRemaining || (result.bytesProduced() > 0)) doUnwrap()
             else flushToUser()
         }
       case CLOSED ⇒
```